File: ml_service/app/routers/health.py
"""Health check router for liveness verification."""

from __future__ import annotations

import importlib.metadata
from typing import Literal

from fastapi import APIRouter
from pydantic import BaseModel

from app.config import get_settings

router = APIRouter()


class HealthResponse(BaseModel):
    """Response schema for the /health endpoint."""

    status: Literal["ok"]
    service: str
    environment: str
    version: str


@router.get("/health", response_model=HealthResponse)
def health_check() -> HealthResponse:
    """Return a liveness signal with service identity and runtime metadata."""
    settings = get_settings()
    try:
        pkg_version = importlib.metadata.version("preventdm-ml-service")
    except importlib.metadata.PackageNotFoundError:
        pkg_version = "0.0.0"
    return HealthResponse(
        status="ok",
        service=settings.app_name,
        environment=settings.app_env,
        version=pkg_version,
    )

File: rust_service/src/routes/health.rs
//! Health check route for liveness verification.

use axum::{extract::State, routing::get, Json, Router};
use serde::Serialize;

use crate::config::Config;

/// Response body returned by `GET /health`.
#[derive(Debug, Serialize)]
pub struct HealthResponse {
    /// Always `"ok"` when the service is running.
    pub status: String,

    /// Human-readable service name from [`Config::app_name`].
    pub service: String,

    /// Deployment environment from [`Config::app_env`].
    pub environment: String,

    /// Package version baked in at compile time via `CARGO_PKG_VERSION`.
    pub version: String,
}

/// Build the health router.
///
/// Registers `GET /health` and expects [`Config`] to be provided as Axum
/// application state via [`Router::with_state`].
pub fn router() -> Router<Config> {
    Router::new().route("/health", get(health_check))
}

async fn health_check(State(config): State<Config>) -> Json<HealthResponse> {
    Json(HealthResponse {
        status: "ok".to_owned(),
        service: config.app_name,
        environment: config.app_env,
        version: env!("CARGO_PKG_VERSION").to_owned(),
    })
}
